src/net_processing.cpp

```diff
@@ -3611,6 +3613,12 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             AddTimeData(pfrom.addr, peer->m_time_offset);
 
             m_outbound_time_offsets.Add(peer->m_time_offset);
+
+            auto median = m_outbound_time_offsets.Median();
+            if (median > TIME_OFFSET_WARN_THRESHOLD.count() ||
+                median < -TIME_OFFSET_WARN_THRESHOLD.count()) {
+                SetMiscWarning(Untranslated("Please check that your computer's date and time are correct"));
```

Comment:
`SetMiscWarning` seems a poor interface; if there's two warnings, the latest one overrides the first, and you can't remove warnings in the event that they resolve themselves. Might be better to do something like `SetfLargeWorkInvalidChainFound(bool)` ?